server/store/mem_store.go

```diff
@@ -0,0 +1,25 @@
+package store
+
+import (
+	"server/model"
+	"sync"
+)
+
+func NewMemStore() *MemStore {
+	return &MemStore{
+		orders: make(map[string]*model.Order),
+	}
+}
+
+// For test purpose
+type MemStore struct {
+	orders map[string]*model.Order
+	lock   sync.Mutex
+}
+
+func (s *MemStore) CreateOrder(o *model.Order) error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	s.orders[o.PaymentID] = o
```

Comment:
Should we consider error case like the order id has existed already?

Reply:
I guess nowpayment will always generate new paymentid.
But this is a good point. Will check this in the next PR.